iac/key_vault.tf

```diff
@@ -0,0 +1,39 @@
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_key_vault" "key_vault" {
+  name                        = local.key_vault_name
+  location                    = azurerm_resource_group.resource_group.location
+  resource_group_name         = azurerm_resource_group.resource_group.name
+  enabled_for_disk_encryption = true
+  tenant_id                   = data.azurerm_client_config.current.tenant_id
+  soft_delete_enabled         = true
+  purge_protection_enabled    = false
+
+  sku_name = "standard"
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = data.azurerm_client_config.current.object_id
```

Comment:
I'm not following. It's giving access to the service principal that deploys this?

Reply:
I believe so, yes. The documentation is a little lacking, but having run this locally I am seeing my user with the given permissions. If I understand the next 2 stories, this will likely be updated to allow/default to the App Service Key Managed Identity?

According to the documentation object_id is required.

https://www.terraform.io/docs/providers/azurerm/r/key_vault.html
https://www.terraform.io/docs/providers/azuread/d/client_config.html

Reply:
Yup. For app service you'll have to do something similar. 

Can you create a keyvault without setting the access policy? Or this based on an example that you were following?

Reply:
Based on the example and documentation, I believe both are required.